packages/flutter/test/material/bottom_sheet_test.dart

```diff
@@ -81,6 +81,41 @@ void main() {
     expect(find.text('BottomSheet'), findsNothing);
   });
 
+  testWidgets('Tapping outside a modal BottomSheet should not dismiss it when isDismissable=false', (WidgetTester tester) async {
+    BuildContext savedContext;
+
+    await tester.pumpWidget(MaterialApp(
+      home: Builder(
+          builder: (BuildContext context) {
+            savedContext = context;
+            return Container();
+          }
+      ),
+    ));
+
+    await tester.pump();
+    expect(find.text('BottomSheet'), findsNothing);
+
+    bool showBottomSheetThenCalled = false;
+    showModalBottomSheet<void>(
+      context: savedContext,
+      builder: (BuildContext context) => const Text('BottomSheet'),
+      isDismissible: false,
+    ).then<void>((void value) {
+      showBottomSheetThenCalled = true;
+    });
+
+    await tester.pumpAndSettle();
+    expect(find.text('BottomSheet'), findsOneWidget);
+    expect(showBottomSheetThenCalled, isFalse);
+
+    // Tap above the bottom sheet to dismiss it
+    await tester.tapAt(const Offset(20.0, 20.0));
+    await tester.pump(); // bottom sheet dismiss animation starts
```

Comment:
```suggestion
    await tester.pump(); // Bottom sheet dismiss animation starts.
```

(I know this is copied from above! Still better to do complete sentences https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences

Reply:
Done.